test/cli-validator/tests/optionHandling.js

```diff
@@ -222,4 +222,70 @@ describe('cli tool - test option handling', function() {
       'Operations must have a non-empty `operationId`.'
     );
   });
+  /*
+  $ lint-openapi -c ./justWarnConfigOverride.json justWarn.yml
+
+errors
+
+  Message :   Schema must have a non-empty description.
+  Path    :   definitions.Category
+  Line    :   137
+
+  Message :   Schema must have a non-empty description.
+  Path    :   definitions.Tag
+  Line    :   149
+
+  Message :   Schema must have a non-empty description.
+  Path    :   definitions.Pet
+  Line    :   161
+
+warnings
+
+  Message :   Response schemas should be defined with a named ref.
+  Path    :   paths./pet/
+   */
+  it('should change output for overridden options when config file is manually specified', async function() {
+    const capturedText = [];
+
+    const unhookIntercept = intercept(function(txt) {
+      capturedText.push(stripAnsiFrom(txt));
+      return '';
+    });
+
+    const program = {};
+    program.args = ['./test/cli-validator/mockFiles/justWarn.yml'];
+    program.default_mode = true;
+    program.config ='./test/cli-validator/mockFiles/justWarnConfigOverrideFull.json';
+
+    const exitCode = await commandLineValidator(program);
+    unhookIntercept();
+
+    expect(exitCode).toEqual(0); // FIXME this should be 1 because there should be an error
+
+    // simple state machine to count the number of warnings and errors.
+    let errorCount = 0;
+    let warningCount = 0;
+    let inErrorBlock = false;
+    let inWarningBlock = false;
+
+    capturedText.forEach(function(line) {
+      if (line.includes('errors')) {
+        inErrorBlock = true;
+        inWarningBlock = false;
+      } else if (line.includes('warnings')) {
+        inErrorBlock = false;
+        inWarningBlock = true;
+      } else if (line.includes('Message')) {
+        if (inErrorBlock) {
+          errorCount++;
+        } else if (inWarningBlock) {
+          warningCount++;
+        }
+      }
+    });
```

Comment:
This is a nifty way of counting the number of errors/warnings in the captured output - nice!

Reply:
does it make sense to put it somewhere shared for other tests to leverage?

Reply:
I was thinking the exact same thing but I think we can wait until it makes sense to add it to another test